rapidfuzz/details/matching_blocks.hpp

```diff
@@ -190,8 +191,8 @@ class SequenceMatcher {
 
 private:
     // Cache to avoid reallocations
-    std::vector<int64_t> j2len_;
-    std::unordered_map<iterator_type<InputIt2>, std::vector<int64_t>> b2j_;
```

Comment:
Is there any reason the replace the usage of `iterator_type` here?

I think the optimum here would be a backport of `std::iter_value_t` which uses `std::iterator_traits<std::remove_cvref_t<T>>::value_type` if possible, but falls back to `std::indirectly_readable_traits<std::remove_cvref_t<T>>::value_type`.

It would be fine with me to define it as:
```
// already exists under the name plain in rapidfuzz-cpp, but should be renamed to match C++20
template <typename T>
using remove_cvref_t = std::remove_cv_t<std::remove_reference_t<T>>;

template<typename T>
using iter_value_t = typename std::iterator_traits<remove_cvref_t<T>>::value_type;
```
for now, which could be improved later, so it:
1) uses `std::iter_value_t` if available
2) closer matches the behavior of `std::iter_value_t`

Reply:
On another thought we need `std::iterator_traits<T>::value_type` to be available for `std::distance` anyways, so we do not need to implement the fallback of `std::iter_value_t`

Reply:
I have not been introduced to this type_traits feature until you mentioned it. Indeed, `std::iter_value_t` is more convenient and can handle more cases than `std::iterator_traits<T>::value_type`. I reverted the change back to `iterator_type`.